### DISCUSSION TITLE
Formatted HTML Display

### DISCUSSION THREAD

**Discussion by raysefo** [QuestPDF/QuestPDF#995] [General]
Hi,
I wonder if it is possible to display formatted HTML on the PDF? If so how?

```
SalesContract = "<b>This is a sales contract. Please read carefully.</b><br>" + 
                               "• Lorem ipsum dolor sit amet, consectetur adipiscing elit.<br>" +
                               "• Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.<br>" +
                               "• Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris.<br>" +
                               "• Nisi ut aliquip ex ea commodo consequat."
```
Here i where I want to display formatted HTML:
```
public class NotesSection : IComponent
{
    private readonly string Notes;

    public NotesSection(string notes)
    {
        Notes = notes;
    }

    public void Compose(IContainer container)
    {
        container.Row(row =>
        {
            row.RelativeItem(350).Background(Colors.Grey.Lighten3).Padding(5).Column(column =>
            {
                column.Spacing(2);
                column.Item().Text("Notlar / Notes:").Bold();
                column.Item().Text(Notes);
            });

            row.ConstantItem(200).Height(75).Border(1).BorderColor(Colors.Grey.Medium).Padding(5).Column(column =>
            {
                column.Spacing(2);
                column.Item().Text($"Yetkili Kişi:");

            });
        });
    }
}
```

**Comment by BeneHenke** [comment]
As far as I'm aware the library does not support any html. Is there a reason you need it as html and not build it with the library?

**Reply by raysefo** [reply]
How can I build this text with the library? let's say I have this text.

`<b>This is a sales contract. Please read carefully.</b><br>• <span style="font-weight: bold;">Lorem </span>ipsum dolor sit amet, consectetur adipiscing elit.<br>• <span style="color: rgb(255, 0, 0);">Sed </span>do eiusmod tempor incididunt ut labore et dolore magna aliqua.<br>• <span style="background-color: rgb(0, 0, 255);">Ut </span>enim ad minim veniam, quis nostrud exercitation ullamco laboris.<br><div style="text-align: right;">• Nisi ut aliquip ex ea commodo consequat.</div>`

**Reply by haraman21** [reply]
> How can I build this text with the library? let's say I have this text.
> 
> `<b>This is a sales contract. Please read carefully.</b><br>• <span style="font-weight: bold;">Lorem </span>ipsum dolor sit amet, consectetur adipiscing elit.<br>• <span style="color: rgb(255, 0, 0);">Sed </span>do eiusmod tempor incididunt ut labore et dolore magna aliqua.<br>• <span style="background-color: rgb(0, 0, 255);">Ut </span>enim ad minim veniam, quis nostrud exercitation ullamco laboris.<br><div style="text-align: right;">• Nisi ut aliquip ex ea commodo consequat.</div>`

HTML is not supported out of the box in this library (yet). You can explore [https://github.com/Relorer/HTMLToQPDF](url)

To directly build this text with the library using fluent API you can try something like:

	column.Item().Text(text =>
	{
		text.Line("This is a sales contract. Please read carefully.").Bold();

		text.Span("• ");
		text.Span("Lorem ").Bold();
		text.Line("ipsum dolor sit amet, consectetur adipiscing elit.");

		text.Span("• ");
		text.Span("Ut ").FontColor(Color.FromRGB(0, 0, 255));
		text.Line("enim ad minim veniam, quis nostrud exercitation ullamco laboris.");

		text.Span("• ");
		text.Span("Ut ").BackgroundColor(Color.FromRGB(0, 0, 255));
		text.Line("enim ad minim veniam, quis nostrud exercitation ullamco laboris.");
	});
	column.Item().AlignRight().Text("• Nisi ut aliquip ex ea commodo consequat.");

**Reply by raysefo** [reply]
My application is a Blazor Server and I am using Radzen HTML Editor. This text is not static, it is just an example. The user can change the color or position etc. I also tried HTMLtoQPDF but since Radzen HTML Editor returns CSS styles in the string like the one in my post using HTMLtoQPDF does not work with the CSS style ones. So I need something more generic to search the text and apply the appropriate fluent API library. But I am not sure how.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Regarding HTML support — we do not plan to fully support all HTML, CSS, and JavaScript features, as doing so would introduce significant security, performance, and resource-consumption concerns. That said, we are considering supporting a subset of HTML with inlined CSS, which is a typical output of WYSIWYG editors. Due to the complexity and risks involved, this is currently in our backlog without a specific timeline. We are actively listening to community demand to help prioritize this effort.

In the meantime, the following community-maintained projects offer support for rendering HTML content within QuestPDF documents:
- [QuestPDF.HTML](https://www.nuget.org/packages/QuestPDF.HTML) (NuGet)
- [HTMLToQPDF](https://github.com/Relorer/HTMLToQPDF) (GitHub)

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
